sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -256,6 +283,22 @@ object EmptyBlock extends Block with Serializable {
   override def + (other: Block): Block = other
 }
 
+/**
+ * A block inlines all types of input arguments into a string without
+ * tracking any reference of `JavaCode` instances.
+ */
+case class InlineBlock(block: String) extends Block {
+  override val code: String = block
+  override val exprValues: Set[ExprValue] = Set.empty
+
+  override def + (other: Block): Block = other match {
+    case c: CodeBlock => Blocks(Seq(this, c))
+    case i: InlineBlock => InlineBlock(block + i.block)
```

Comment:
does this mean that if we have to consecutive `InlineBlock`s we are concatenating them without any space? Is this ok?

Reply:
I think we won't have too many cases of concatenating `InlineBlock`. You can see `InlineBlock` is mostly used to wrap a small piece of code like a java class name. I'm not sure if we need to add a newline here.

Reply:
That said, I don't think we should have an usage like:
```
val inline1 = InlineBlock(".....")
val inline2 = inline1 + InlineBlock(".....")
code"""
     | $inline2
     | ...
   """.stripMargin
```

Reply:
as of now probably we don't have... @cloud-fan @kiszk what do you think?